src/metabase/db.clj

```diff
@@ -49,7 +49,10 @@
   "Parse a DB connection URI like `postgres://cam@localhost.com:5432/cams_cool_db?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory` and return a broken-out map."
   [uri]
   (when-let [[_ protocol user pass host port db query] (re-matches #"^([^:/@]+)://(?:([^:/@]+)(?::([^:@]+))?@)?([^:@]+)(?::(\d+))?/([^/?]+)(?:\?(.*))?$" uri)]
-    (merge {:type     (keyword protocol)
+    (merge {:type     (case (keyword protocol)
```

Comment:
We probably wouldn't need to parse those because we actually use connection strings under the hood for the drivers themselves. We could just pass them directly to JDBC